packages/contracts/contracts/libs/v0.8.x/minter-libs/PolyptychLib.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../../../interfaces/v0.8.x/IGenArt721CoreContractV3_Base.sol";
+import "../../../interfaces/v0.8.x/IGenArt721CoreContractExposesHashSeed.sol";
+import "../../../interfaces/v0.8.x/ISharedRandomizerV0.sol";
+
+import "@openzeppelin-4.7/contracts/token/ERC20/IERC20.sol";
+
+pragma solidity ^0.8.0;
+
+/**
+ * @title Core contract interface for accessing the randomizer from the minter
+ * @notice This interface provides the minter with access to the shared
+ * randomizer, allowing the token hash seed for a newly-minted token to be
+ * assigned by the minter if the artist has enabled the project as a polyptych.
+ * Polytptych projects must use the V3 core contract, this polyptych minter,
+ * and a shared randomizer - this interface allows the minter to access the
+ * randomizer.
+ */
+interface IGenArt721CoreContractV3WithSharedRandomizer is
```

Comment:
I think the main reason for (originally) defining it here is no requirement that a core contract has a _shared_ randomizer assigned, therefore it is the minter's assumption that the core contract has a shared randomizer assigned (vs. a non-shared randomizer). It wouldn't make sense to have the core contracts conform to this interface, because they aren't guaranteed to be returning a shared randomizer.

That being said, if we end up with multiple minters assigning hash seeds, this would make sense to break out. Just because it is currently used only by this minter doesn't mean we shouldn't break it out, so I'll go ahead and define it it its own file 👍

Reply:
split out in [3575a28](https://github.com/ArtBlocks/artblocks-contracts/pull/838/commits/3575a284c6809a840f1f8377070b1650873dd645)